### PR TITLE
Importing/exporting of assembly frames from/to text

### PR DESCRIPTION
**Pull request by Exxion**
Please fucking kill me this was awful

Sorry here's what I actually did:
Assembly frames now have seven new procs. (Wait calm down keep reading.) One just handles insertion of assemblies into the frame. It's basically the same as before except it doesn't assume a mob is holding the assembly and it's handled outside of attackby(), which now calls this proc to insert assemblies.
Two of the remaining six are for debug purposes; I'll explain them at the end, but they're pretty simple.
Two of the others are mainly meant to be used by one of the others. You can read the code to see exactly what they do.

The remaining two are the real part of this PR.
The first is to_text(). It simply returns a somewhat human-readable string containing the information about the assemblies inside (well, all the information that can be accessed by the assembly frame, at least).
The other is from_text(). This is the COOL part of this PR.
from_text() accepts three args. The first is the text that to_text() outputs, or at least a string of that form. Given no other args, the assembly frame assembles itself from the text given, just creating assemblies as needed.
If the second arg is true, the third arg must be given. The third arg is a list of assemblies available to the assembly frame to set itself up as the text data dictates. If the list contains the assemblies the frame requires, it will take and configure them, preserving the order of individual assemblies of a type (not types of assemblies, which are in the order dictated by the text data) in the list from which they were taken. The preservation of the order is important for things that have variables not in accessible_values, such as light tile controllers. (If the list does not contain the required assemblies, nothing really happens.)

I left in the two debug procs because they're the only real way to use this for now. I'm too tired and done with this fucking shit to code a proper in-game way to use this feature at the moment. For the time being, admins can fuck around with debug_to_text() and debug_from_text().

debug_to_text() spawns a piece of paper containing the output of to_text() on the assembly frame's turf.
debug_from_text() accepts one arg. It defaults to false. If left false, it just calls from_text() with the data off the first piece of paper it finds in the same loc as the assembly frame. If that arg is set to true, it calls from_text() with the data of the first paper it finds in the loc and uses the first cardboard box (of any kind) it finds in that loc as the list of assemblies to draw from.

Anyone who wants to is ~~invited~~ encouraged to code a machine or something that gives players an actual way to use this feature.

Fully tested for normal use, at least. It might be possible to break it if you try to do so, but I _think_ I did a pretty good job on the validation.

### CODE REVIEW

**Comment by clusterfack** [comment]
tl;dr

**Comment by unid15** [comment]
:+1: Great job! Could save tons time in the future seeing as some assembly frame setups can take hours to build

**Comment by clusterfack** [comment]
Just want to say that I hope you both tested this and thoroughly checked it because the code is too in depth to gain anything from my code review.

**Comment by Exxion** [comment]
Well, it properly sets the label, accessible_values, and connections for assemblies with either use_parts = 0 or 1, as long as the text data is valid (and there are enough parts if use_parts = 1). I'm quite confident that the validation is sufficient that fucking with the text you give to from_text() will either work properly, not work at all,  or only make the assembly frame not work how you wanted and not break anything else. Well... unless you're an admin and include a broken assembly type in the text data. This won't be a problem for regular players since use_parts = 1 means they would have to have the assembly for it to be inserted into the frame, but the assembly frame could summon a base assembly or an IR tripwire if asked to with use_parts = 0. Since I only expect it to be called with use_parts = 0 by admins, though, I don't see this as a problem.
